core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java

```diff
@@ -632,7 +632,7 @@ private void requestTabletHosting(ClientContext context,
     }
 
     // System tables should always be hosted
-    if (AccumuloTable.ROOT.tableId() == tableId || AccumuloTable.METADATA.tableId() == tableId) {
+    if (AccumuloTable.allTableIds().contains(tableId)) {
```

Comment:
I think the class `AccumuloTable` is really a misnomer. It's really `TablesInTheAccumuloNamespace`, and would probably be more intuitive if named `AccumuloNamespace`, with methods like `.containsTable(tableId)`, or `.tables().contains(TableId)`. There are several places in the code that have this pattern of checking that a tableId is contained in this set that would benefit from such a change. A similar comment was made on #5451

Reply:
Created https://github.com/apache/accumulo/pull/5498